### ISSUE TITLE
Preliminary results: somatic growth

### ISSUE THREAD

**Issue by ebuhle** [ebuhle/LeatherbackNesting#4]
Well, @hoffmannsarahlouise, here's another round of somewhat underwhelming results. I analyzed the size data by calculating specific growth rate (SGR) in maximum CCL (`log(final / initial) / elapsed`, where elapsed time is measured in decades), over each observation interval for all turtles that were encountered more than once. Thus if an individual was encountered more than twice, she will have multiple observed growth increments.

I then modeled SGR using, again, hierarchical Bayesian regression models fitted in **rstanarm**. The base model includes random turtle- and year-specific intercepts. Model 2 adds a linear effect of initial size; this is a cheap way of approximating a non-(log)-linear growth function, given that we don't have enough longitudinal time series to fit a theoretical model such as the von Bertalanffy or whatever. Model 3 includes turtle-specific slopes as well as intercepts. Finally, Model 4 adds a linear time trend.

I'll cut to the chase: there is strong evidence that SGR declines as turtles grow larger (the `ccl_max0_std` effect below, which is not surprising; constant SGR would imply exponential growth in size) but no indication of a systematic trend in growth over time (`year_ctr`). There is some turtle-level heterogeneity in baseline growth (error terms: intercept grouped by name) and the slope of size-dependence (`ccl_max0_std` effect grouped by name), and some random interannual fluctuations (intercept grouped by `fyear`), but overall residual error dominates.

```r
stan_lmer
 family:       gaussian [identity]
 formula:      sgr ~ ccl_max0_std + year_ctr + ((1 | name) + (0 + ccl_max0_std | name)) + (1 | fyear)
 observations: 324
------
             Median MAD_SD
(Intercept)   0.035  0.006
ccl_max0_std -0.016  0.003
year_ctr     -0.002  0.001

Error terms:
 Groups   Name         Std.Dev.
 name     (Intercept)  0.01067 
 name     ccl_max0_std 0.01187 
 fyear    (Intercept)  0.01776 
 Residual              0.04346 
Num. levels: name 207, fyear 10 
``` 

Once again, comparing the marginal distribution of the observations (black) to 100 hypothetical replicate data sets simulated from the posterior predictive distribution reveals that the model leaves something to be desired:

<img src=https://user-images.githubusercontent.com/22503560/118180411-59c98e80-b3eb-11eb-91fc-02a027f97267.png width="60%" height="60%">

This appears to be driven by a very long-tailed distribution of turtle-specific intercepts, which is to say that some turtles grew (or shrank!) much more than others on average:

<img src=https://user-images.githubusercontent.com/22503560/118180574-8a112d00-b3eb-11eb-900e-8baf8f923e6d.png width="50%" height="50%">

We can get a better sense of the variation by looking at the individual growth trajectories. Indeed, it looks like a few females shrank considerably while others grew much faster than average:

<img src=https://user-images.githubusercontent.com/22503560/118181101-33f0b980-b3ec-11eb-9c4e-e006ab169363.png width="60%" height="60%">

Here is the model fit showing the non-(log)-linearity in growth. The inner shading around the fitted line indicates the uncertainty in the hyperparameters (i.e., the overall average fit) and the light outer shading additionally incorporates the turtle-level heterogeneity in the intercept and slope as well as the interannual fluctuation. You can roughly interpret SGR as the proportional growth rate, so on average turtles are growing around 3% per decade, but some individuals grew or shrank by 20% or more.

<img src=https://user-images.githubusercontent.com/22503560/118181544-bbd6c380-b3ec-11eb-800c-29a95adf2154.png width="60%" height="60%">

Would love to hear your thoughts on this! I also looked at body size of females that were only encountered once and didn't see any obvious patterns or trends over time, so I'm putting that one aside for the time being to focus on the proportion of neophytes and nest success.

**Comment by hoffmannsarahlouise** [comment]
Hi Eric,

This is really interesting! I’m pretty shocked to see turtles are shrinking, I’m not sure what to think of that. I’m also surprised to see no trend in time of nesting.

I have to admit, most of this analysis is pretty far over my head. Once you’ve had a chance to look through the neophyte data, could we set up a meeting to talk about the implications of the models and what that would realistically mean from an ecological standpoint?

Thanks for working on this!

**Comment by ebuhle** [comment]
> Once you’ve had a chance to look through the neophyte data, could we set up a meeting to talk about the implications of the models and what that would realistically mean from an ecological standpoint?

Yes, for sure. I'm aiming to get a draft vignette done by this weekend or early next week, which will put everything in one place and provide a jumping-off point for discussion. In the meantime, I would welcome any comments on the results shown in these threads, especially if you see anything that looks surprising or "off". I've tried to emphasize visualizations that show the patterns in the raw data, because in many ways it's looking like that may be one of the most useful outcomes of this initial analysis. 

I was also going to ask whether any of your Marinelife colleagues are on GitHub. I've found GH Issues to be a great forum for discussing analyses-in-progress with collaborators, even / especially those who are less technically inclined. It's more interactive than a rendered RMarkdown report but more structured than email, and the ability to link to code and results in the repo is really key. Anyway, feel free to direct the Marinelife folks to these Issues and share feedback (ideally as comments here)!
